src/mds/mdstypes.h

```diff
@@ -458,7 +458,12 @@ struct inode_t {
   }
 
   // file type
+  #ifndef _WIN32
   bool is_symlink() const { return (mode & S_IFMT) == S_IFLNK; }
+  #else
+  // TODO: provide proper implementation. We'll need a file name or handle.
+  bool is_symlink() const { return 0; }
+  #endif
```

Comment:
Same

Reply:
I was a bit concerned that the Windows stat function doesn't set S_IFLNK for symlinks but probably that can be handled somewhere else.

Reply:
The thing is that CephFS reuses the POSIX API for encoding `st_mode` in the CephFS inode struct. This made the code simpler at the time but makes your life more difficult when porting Ceph to windows. The MDS is not using the Windows `stat` function (because CephFS file systems do not exist on the local file system!) but it does use all of the related macros.

Reply:
See also: https://tracker.ceph.com/issues/21058